examples/ten_stage_analysis/src/main.rs

```diff
@@ -0,0 +1,67 @@
+use rb_link::*;
+use std::thread;
+use std::time::Duration;
+
+fn main() {
```

Comment:
Since fifo blocking analyse is a very very common use case, so the related function and algroithm should be included in your framework and provide reusable API to users.

Besides, when move the function to your lib, do not include all code in a single lib.rs